components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyConnection.java

```diff
@@ -52,23 +58,14 @@
      *
      * @param origin  the origin connected to
      * @param channel the netty channel used
+     * @param factory used to create operation objects that send http requests via this connection
      */
     @VisibleForTesting
-    public NettyConnection(Origin origin, Channel channel) {
-        this(new UUID(), origin, channel);
-    }
-
-    /**
-     * Constructs an instance.
-     *
-     * @param id      an object to use as an ID
-     * @param origin  the origin connected to
-     * @param channel the netty channel used
-     */
-    public NettyConnection(Object id, Origin origin, Channel channel) {
-        this.id = id;
+    public NettyConnection(Origin origin, Channel channel, HttpRequestOperationFactory factory) {
+        this.id = new UUID();
         this.origin = checkNotNull(origin);
         this.channel = checkNotNull(channel);
+        this.requestOpFactory = factory::newHttpRequestOperation;
```

Comment:
Why not just use `HttpRequestOperationFactory` as the field type, so that this method reference isn't required?